tools/js-tools/eslintrc/base.mjs

```diff
@@ -325,6 +325,18 @@ export default function makeBaseConfig( configurl, opts = {} ) {
 					'error',
 					{ allowInterfaces: 'with-single-extends' },
 				],
+				// Disable param name check for destructured params.
+				'jsdoc/check-param-names': [ 'error', { checkDestructured: false } ],
```

Comment:
Why are you disabling this? jsdoc's `recommended-typescript` config doesn't do this.

What this seems to disable is it warning about the missing `@param opts.foo` and incorrect `@param opts.bar` in something like this:
```ts
/**
 * @param opts - Options
 * @param opts.bar - Bar
 * @returns xxx
 */
function xxx( { foo: string } ): null {
    return null;
}
```
neither of which seem to have anything to do with types.

Reply:
It disables it only when the props are restructured, e.g. in component props; otherwise, it maintains the default behavior.

Reply:
Otherwise this will have that error

```ts
/**
 * Desc.
 *
 * @param opts - Options
 *
 * @return xxx
 */
function xxx( { foo: string } ): null {
	return null;
}
```

<img width="662" alt="image" src="https://github.com/user-attachments/assets/d5f8a969-0bc4-49eb-80f0-78d9582f8541" />

Reply:
Why shouldn't `opts.foo` be documented there?

Reply:
Because I can and would prefer to document it like this

```ts
type FooProps = {
	/**
	 * Description for foo.
	 */
	foo: string;
};

/**
 * Desc.
 *
 * @param props - Component props.
 *
 * @return Rendered Foo
 */
function Foo( { foo: string }: FooProps ): null {
	return null;
}
```

Reply:
But will everyone else? Or will they skip it even when there's no `FooProps` type involved?

I note that adding `{FooProps}` to the `@param` makes it do what you want:
```ts
/**
 * Desc.
 *
 * @param {FooProps} props - Component props.
 *
 * @return Rendered Foo
 */
function Foo( { foo }: FooProps ): null {
	return null;
}
```
This is due to the default setting of `checkTypesPattern`. At a quick search I didn't find a setting for doing the same when the type is omitted in `@param` because there's a TypeScript type on the parameter though. That might be a useful feature request for eslint-plugin-jsdoc.

Reply:
Alright, I have removed that for now.